govcd/nat_test.go

```diff
@@ -0,0 +1,142 @@
+
+	vnicIndex, err := edge.GetVnicIndexByNetworkNameAndType(vcd.config.VCD.ExternalNetwork, "uplink")
+	check.Assert(err, IsNil)
+
+	natRule := &types.EdgeNatRule{
+		Action:            "dnat",
+		Vnic:              vnicIndex,
+		Protocol:          "tcp",
+		OriginalAddress:   vcd.config.VCD.ExternalIp,
+		OriginalPort:      "443",
+		TranslatedAddress: vcd.config.VCD.InternalIp,
+		TranslatedPort:    "8443",
+		Enabled:           true,
+		LoggingEnabled:    true,
+		Description:       "my description",
+	}
+	if testVerbose {
+		fmt.Printf("# %s %s %s:%s -> %s:%s\n", natRule.Action, natRule.Protocol, natRule.OriginalAddress,
+			natRule.OriginalPort, natRule.TranslatedAddress, natRule.TranslatedPort)
+	}
+
+	testNsxvNat(natRule, vcd, check, edge)
+
+	natRule = &types.EdgeNatRule{
+		Action:            "dnat",
+		Vnic:              vnicIndex,
+		Protocol:          "icmp",
+		IcmpType:          "router-advertisement",
+		OriginalAddress:   vcd.config.VCD.ExternalIp,
+		TranslatedAddress: vcd.config.VCD.InternalIp,
+		Enabled:           true,
+		LoggingEnabled:    true,
+		Description:       "my description",
+	}
+	if testVerbose {
+		fmt.Printf("# %s %s:%s %s -> %s\n", natRule.Action, natRule.Protocol, natRule.IcmpType,
+			natRule.OriginalAddress, natRule.TranslatedAddress)
+	}
+	testNsxvNat(natRule, vcd, check, edge)
+
+	natRule = &types.EdgeNatRule{
+		Action:            "dnat",
+		Vnic:              vnicIndex,
+		Protocol:          "any",
+		OriginalAddress:   vcd.config.VCD.ExternalIp,
+		TranslatedAddress: vcd.config.VCD.InternalIp,
+		Enabled:           true,
+		LoggingEnabled:    true,
+		Description:       "my description",
+	}
+	if testVerbose {
+		fmt.Printf("# %s %s %s -> %s\n", natRule.Action, natRule.Protocol, natRule.OriginalAddress,
+			natRule.TranslatedAddress)
+	}
+	testNsxvNat(natRule, vcd, check, edge)
+}
+
+// testNsxvNat is a helper to test multiple configurations of NAT rules. It does the following
+// 1. Creates NAT rule with provided config
+// 2. Checks that it can be retrieve and verifies if IDs match
+// 3. Tries to update description field and validates that nothing else except description changes
+// 4. Deletes the rule
+// 5. Validates that the rule was deleted
+func testNsxvNat(natRule *types.EdgeNatRule, vcd *TestVCD, check *C, edge EdgeGateway) {
+	createdNatRule, err := edge.CreateNsxvNatRule(natRule)
+	check.Assert(err, IsNil)
+
+	parentEntity := vcd.org.Org.Name + "|" + vcd.vdc.Vdc.Name + "|" + vcd.config.VCD.EdgeGateway
+	AddToCleanupList(createdNatRule.ID, "nsxvNatRule", parentEntity, check.TestName())
+
+	gotNatRule, err := edge.GetNsxvNatRuleById(createdNatRule.ID)
+	check.Assert(err, IsNil)
+	check.Assert(gotNatRule, DeepEquals, createdNatRule)
+	check.Assert(gotNatRule.ID, Equals, createdNatRule.ID)
+
+	// Set ID and update nat rule with description
+	natRule.ID = gotNatRule.ID
+	natRule.Description = "Description for NAT rule"
+	updatedNatRule, err := edge.UpdateNsxvNatRule(natRule)
```

Comment:
`updateNatRule` missing check it isn't nil

Reply:
Done.